examples/nlp/duplex_text_normalization/duplex_text_normalization_train.py

```diff
@@ -0,0 +1,99 @@
+"""
+This script contains an example on how to train a DuplexTextNormalizationModel.
+Note that DuplexTextNormalizationModel is essentially a wrapper class around
+two other classes:
+
+(1) DuplexTaggerModel is a model for identifying spans in the input that need to
+be normalized. Usually, such spans belong to semiotic classes (e.g., DATE, NUMBERS, ...).
+
+(2) DuplexDecoderModel is a model for normalizing the spans identified by the tagger.
+For example, in the text normalization (TN) problem, each span will be converted to its
+spoken form. In the inverse text normalization (ITN) problem, each span will be converted
+to its written form.
+
+Therefore, this script consists of two parts, one is for training the tagger model
+and the other is for training the decoder.
+
+This script uses the `/examples/nlp/duplex_text_normalization/conf/duplex_tn_config.yaml`
+config file by default. The other option is to set another config file via command
+line arguments by `--config-name=CONFIG_FILE_PATH'. Probably it is worth looking
+at the example config file to see the list of parameters used for training.
+
+USAGE Example:
+1. Obtain a processed dataset (refer to the `text_normalization doc <https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization.rst>`)
+2.
+# python duplex_text_normalization_test.py
+        data.base_dir=PATH_TO_DATASET_DIR
+        mode={tn,itn,joint}
+
+There are 3 different modes. `tn` mode is for training a system for TN only.
+`itn` mode is for training a system for ITN. `joint` is for training a system
+that can do both TN and ITN at the same time. Note that the above command will
+first train a tagger and then train a decoder sequentially.
+
+You can also train only a tagger (without training a decoder) by running the
+following command:
+# python duplex_text_normalization_test.py
+        data.base_dir=PATH_TO_DATASET_DIR
+        mode={tn,itn,joint}
+        decoder_model.do_training=false
+
+Or you can also train only a decoder (without training a tagger):
+# python duplex_text_normalization_test.py
+        data.base_dir=PATH_TO_DATASET_DIR
+        mode={tn,itn,joint}
+        tagger_model.do_training=false
+
+Information on the arguments:
+
+Most arguments in the example config file are quite self-explanatory (e.g.,
+`decoder_model.optim.lr` refers to the learning rate for training the decoder).
+Some arguments we want to mention are:
+
++ data.base_dir: The path to the dataset directory. It is expected that the
+directory contains three files: train.tsv, dev.tsv, and test.tsv. 
```

Comment:
since this is not the original dataset format of google dataset, could you please add data processing script into this directory to prepare data for model?

Reply:
Yes. I added `google_data_preprocessing.py` to the folder.